src/Icicle/Source/ToCore/ToCore.hs

```diff
@@ -646,16 +580,28 @@ convertReduce xx
 
         scrut' <- convertReduce scrut
         pats'  <- convertCaseFreshenPats pats
-        alts'  <- mapM convertExp alts
 
-        let bs' = fst scrut'
+        -- Because the alternatives can contain more folds and stuff, we need to
+        -- use convertReduce on them.
+        -- However, because the pattern variables must be Aggregates, we know that
+        -- any pattern variables will only be mentioned in the postcomputations.
+        -- Therefore we need to pull out the postcomputations into a let,
+        -- and stick them inside the new case alternative.
+        --
+        -- All the foldy bits of each alternative must be computed, because
+        -- we won't know which ones will be needed until after they are run.
```

Comment:
I'm not sure how there can be no pattern variables: what does
```
case x
| Left i -> …
| Right j -> …
```
become?

the reasoning here, anyway, is that if you have
```
case Left (sum value)
| Left a -> (mean value) * a
| Right b -> (min value) * b
```
then the postcomputations of the alternatives will actually just be something like "comp$mean$value * a" and "comp$min$value * b", so you need to pull them back inside the actual case statement underneath where "a" and "b" are bound